### ISSUE TITLE
improve inscription page accessibility

### ISSUE THREAD

**Issue by birkin** [Brown-University-Library/usep-data#391]
## Birkin intro

This is another LLM suggestion. Hope it's useful!

---

 
 ## Likely source of the accessibility issue

  The live inscription page at:

  `https://library.brown.edu/projects/usep/inscription/CA.Berk.UC.HMA.L.8-2330/`

  appears to be rendered from `resources/xsl/USEp_display.xsl`.

  Evidence:
  - The Django app passes `/static/resources/xsl/USEp_display.xsl` into the inscription page as the transform URL.
  - The live accessibility failures match markup emitted directly by `USEp_display.xsl`.
  - Specifically, the stylesheet generates the XML source icon link near the bottom of the file.

  ## Live severe/red accessibility issues

  The severe issues on the live page are:

  - `image-alt`
  - `link-name`

  These come from the XML source link/icon generated by `USEp_display.xsl`.

  Current pattern in the stylesheet:

  ```xsl
  <a href="{concat('https://github.com/Brown-University-Library/usep-data/blob/master/xml_inscriptions/transcribed/',/t:TEI/t:teiHeader/t:fileDesc/
  t:publicationStmt/t:idno/@xml:id,'.xml')}">
      <img style="height:50px;" src="{concat($imageDir, '/xmlIcon.png')}"/>
  </a>
```

  Equivalent blocks also exist for metadata_only and bib_only.

  This outputs:

  - an <img> with no alt
  - an <a> with no text and no accessible name

  ## Suggested stylesheet fix

  Please give the XML source link an accessible name, and give the image meaningful alt text or make it decorative.

  ### Recommended approach

  For all three XML-source blocks:

  - add an aria-label and/or title on the <a>
  - add meaningful alt text on the <img>

  ### Suggested replacement pattern

  ```xsl
  <xsl:variable name="inscription-id"
      select="/t:TEI/t:teiHeader/t:fileDesc/t:publicationStmt/t:idno/@xml:id"/>

  <a
      href="{concat('https://github.com/Brown-University-Library/usep-data/blob/master/xml_inscriptions/transcribed/', $inscription-id, '.xml')}"
      aria-label="{concat('View source XML for ', $inscription-id)}"
      title="{concat('View source XML for ', $inscription-id)}"
  >
      <img
          style="height:50px;"
          src="{concat($imageDir, '/xmlIcon.png')}"
          alt="{concat('View source XML for ', $inscription-id)}"
      />
  </a>
```

  Do the same for the metadata_only and bib_only branches.

  ## Optional cleaner alternative

  If the visible icon is purely decorative, another acceptable approach is:

  - keep the accessible name on the <a>
  - set the image alt to empty

  Example:

  ```xsl

  <a
      href="{...}"
      aria-label="{concat('View source XML for ', $inscription-id)}"
      title="{concat('View source XML for ', $inscription-id)}"
  >
      <img
          style="height:50px;"
          src="{concat($imageDir, '/xmlIcon.png')}"
          alt=""
      />
  </a>
```

  That still fixes the red errors because the link now has a discernible name.

  ## Additional non-red issue in the same stylesheet

  There is also a heading-order issue caused by:

 ` <h3>Summary</h3>`

  appearing before sections rendered as <h2> later in the page.

  That is not one of the severe/red failures, but if desired it could be improved by changing Summary, TM Number, Transcription, Attributes, Credit, and Bibliography to a consistent heading hierarchy.

---

**Comment by emylonas** [comment]
Should be fixed. pls check and close

**Comment by birkin** [comment]
@emylonas and team, Checked [inscription page][ins] -- looks great; no red WAVE-extension errors!

Thanks, closing this issue.

[ins]: <https://library.brown.edu/projects/usep/inscription/CA.Berk.UC.HMA.G.8-3898/>
